mlrun/runtimes/pod.py

```diff
@@ -411,6 +420,264 @@ def enrich_resources_with_default_pod_resources(
+        """
+        # both needs to exist to prune required affinity from spec affinity
+        if not self.affinity or not node_selector_requirements:
+            return
+        if self.affinity.node_affinity:
+            node_affinity: client.V1NodeAffinity = self.affinity.node_affinity
+
+            new_required_during_scheduling_ignored_during_execution = None
+            # we are only looping over required_during_scheduling_ignored_during_execution
+            # because the scheduler can't schedule the Pod unless the rule is met
+            if node_affinity.required_during_scheduling_ignored_during_execution:
+                node_selector: client.V1NodeSelector = (
+                    node_affinity.required_during_scheduling_ignored_during_execution
+                )
+                new_node_selector_terms = self._get_node_selector_terms_without_provided_node_selector_requirements(
+                    node_selector_terms=node_selector.node_selector_terms,
+                    node_selector_requirements_to_remove=node_selector_requirements,
+                )
+                # check whether there are node selector terms to add to the new list of required terms
+                if len(new_node_selector_terms) > 0:
+                    new_required_during_scheduling_ignored_during_execution = (
+                        client.V1NodeSelector(
+                            node_selector_terms=new_node_selector_terms
+                        )
+                    )
+            # check if there is something to set into required_during_scheduling_ignored_during_execution
+            if (
+                not node_affinity.preferred_during_scheduling_ignored_during_execution
+                and not new_required_during_scheduling_ignored_during_execution
+            ):
+                self.affinity.node_affinity.required_during_scheduling_ignored_during_execution = (
+                    None
+                )
+                return
+
+            self._initialize_affinity()
+            self._initialize_node_affinity()
+            self.affinity.node_affinity.required_during_scheduling_ignored_during_execution = (
+                new_required_during_scheduling_ignored_during_execution
+            )
+
+    @staticmethod
+    def _get_node_selector_terms_without_provided_node_selector_requirements(
+        node_selector_terms: typing.List[client.V1NodeSelectorTerm],
+        node_selector_requirements_to_remove: typing.List[
+            client.V1NodeSelectorRequirement
+        ],
+    ) -> typing.List[client.V1NodeSelectorTerm]:
+        """
+        Goes over each expression in all the terms provided and removes the expressions if it matches
+        one of the requirements provided to remove
+
+        :return: New list of terms without the provided node selector requirements
+        """
+        new_node_selector_terms: typing.List[client.V1NodeSelectorTerm] = []
+        # for all term expressions on function spec
```

Comment:
You're getting it from outside and you don't know if it's from the function spec or not, in general the code line below is pretty straight forward IMO, I would remove the comment